private/bufpkg/bufmodule/bufmodulebuild/bufmodulebuild.go

```diff
@@ -168,3 +169,9 @@ func WithExcludePathsAllowNotExist(excludePaths []string) BuildOption {
 		buildOptions.pathsAllowNotExist = true
 	}
 }
+
+// IsErrDuplicatedDependency checks if the given error is `errDuplicatedDependency`, from the build
+// process.
+func IsErrDuplicatedDependency(err error) bool {
```

Comment:
Open for suggestions on naming, this seemed appropriate.

Reply:
Call the error `ErrDuplicateDependency`, and just make it public so people can use `errors.Is`

Reply:
If the error is public `var err = errors.New(...)`, isn't that leaving it open for users to modify it? That's why I didn't exported it, just like in [here](https://github.com/bufbuild/buf/blob/3727d810c3f1de8bc1b39c7bc53b4b9fc6b83c38/private/pkg/storage/errors.go#L40-L43)... although that pkg also has other exported errors.

I'll export it anyway.

Reply:
I think by convention, public errors are expected not to be edited - even in that same file, we export some errors. The `IsNotExist` function is really to match `os.IsNotExist`, which a lot of people have traditionally used.

I actually like the `Is` pattern, and we have used it in `buf` in the past a decent amount, but my impression is `errors.Is` is the golang-idiomatic way forward nowadays.